doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/SiteRenderingContext.java

```diff
@@ -260,9 +279,9 @@ public void setTemplateName(String templateName) {
      * Add a site directory, expected to have a Doxia Site layout, ie one directory per Doxia parser module containing
      * files with parser extension. Typical values are <code>src/site</code> or <code>target/generated-site</code>.
      *
-     * @param siteDirectory a {@link java.io.File} object.
+     * @param siteDirectory a {@link SiteDirectory} object.
      */
-    public void addSiteDirectory(File siteDirectory) {
+    public void addSiteDirectory(SiteDirectory siteDirectory) {
```

Comment:
Why breaking backwards compatibility? Just leave the old method taking `File` in place, deprecate it and just delegate to the new `addSiteDirectory` For the new method please add `@since`.

Reply:
The general question: What is the affected audience? The only known consumer/caller is Maven Site Plugin. You expect this to break in a combination you can already forsee?

Reply:
Will do.

Reply:
Done